File: web/src/types/leaderboard.ts
export interface LeaderboardEntry {
    rank: number;
    user_id: number;
    username: string;
    display_name: string;
    total_power: number;
}

export interface LeaderboardResponse {
    /** Type réellement affiché quand le serveur l'a choisi. */
    type_name?: string | null;
    entries: LeaderboardEntry[];
}

File: backend/app/schemas/leaderboard.py
"""
Schemas — classements de puissance (amis, global, par type de personnage).
"""

from pydantic import BaseModel


class LeaderboardEntry(BaseModel):
    rank: int
    user_id: int
    username: str
    display_name: str
    total_power: int


class LeaderboardResponse(BaseModel):
    # Type réellement affiché, quand le serveur l'a choisi lui-même.
    type_name: str | None = None
    entries: list[LeaderboardEntry] = []
